modules/avatar/avatar.go

```diff
@@ -22,8 +23,9 @@ import (
 	_ "golang.org/x/image/webp" // for processing webp images
 )
 
-// AvatarSize returns avatar's size
-const AvatarSize = 290
+// DefaultAvatarSize is used for avatar generation, usually the avatar image saved in server won't be larger than this value.
+// Unless the original file is smaller than the resized image.
+const DefaultAvatarSize = 256
```

Comment:
This means the profile avatar will not be integer-scaled anymore. Not that it really matters though.

Reply:
TBH, I do not think the old code really makes sense. 

In most cases, the avatar is only display as 16x16 or 28x28, at most maybe 40x40?

Only on the profile, the avatar is displayed as 260x260 (the size is controlled by the width:100% from parent container)

----

So I think Gitea doesn't need to do too much resizing trick.